File: user/src/main/java/com/epam/esm/utils/ampq/MessageListener.java
package com.epam.esm.utils.ampq;

import com.epam.esm.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class MessageListener {
    private final UserService userService;

    @RabbitListener(queues = "${user.deletion.queue}")
    public void onUserDeletion(String email) {
        userService.delete(email);
    }

    @RabbitListener(queues = "${user.creation.queue}")
    public void onUserCreation(CreateUserRequest request) {
        userService.create(request);
    }

    @RabbitListener(queues = "${user.image.response.queue}")
    public void onUserCreation(ImageUploadResponse response) {
        userService.setUploadedImage(response);
    }
}

File: notification/src/main/kotlin/com/epam/esm/utils/amqp/listeners/MessageListener.kt
package com.epam.esm.utils.amqp.listeners

import com.epam.esm.mail.EmailService
import com.epam.esm.utils.amqp.EmailValidationMessage
import com.epam.esm.utils.amqp.PurchaseCreationMessage
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Service

@Service
class MessageListener(val emailService: EmailService) {
    @RabbitListener(queues = ["\${purchase.queue}"])
    fun onOrderCreation(message: PurchaseCreationMessage) {
        emailService.sendOrderConfirmationEmail(message.email, message.purchaseDTO)
    }

    @RabbitListener(queues = ["\${user.queue}"])
    fun onUserRegister(message: EmailValidationMessage) {
        emailService.sendEmailVerificationEmail(message.email, message.activationUrl)
    }
}